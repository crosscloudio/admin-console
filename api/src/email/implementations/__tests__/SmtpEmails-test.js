jest.mock('nodemailer');

describe('SmtpEmails', () => {
  beforeEach(() => {
    Date.now = jest.fn(
      // Tue May 30 2017 10:40:16 GMT+0200 (CEST)
      () => 1496133616539
    );
  });

  describe('#sendResetPasswordEmail', () => {
    it('should sent email', async () => {
      jest.resetModules();

      process.env.SMTP_URL = 'smtp://localhost:4444/?secure=false';
      process.env.SMTP_SENDER_EMAIL = 'me@gmail.com';
      delete process.env.MAILJET_API_KEY;
      delete process.env.MAILJET_API_SECRET;
      delete process.env.MAILJET_SENDER_EMAIL;

      const SmtpEmails = require(// eslint-disable-line global-require
      '../SmtpEmails').default;

      const helper = new SmtpEmails();
      const fakeUser = {
        id: 1,
        name: 'John Smith',
        email: 'sample@gmail.com',
      };

      const nodemailer = require('nodemailer'); // eslint-disable-line global-require

      // `Set password` email
      await helper.sendResetPasswordEmail(fakeUser, { initial: true });
      expect(nodemailer.getSentMails()).toMatchSnapshot();

      // `Reset password` email
      await helper.sendResetPasswordEmail(fakeUser);
      expect(nodemailer.getSentMails()).toMatchSnapshot();
    });
  });
});
