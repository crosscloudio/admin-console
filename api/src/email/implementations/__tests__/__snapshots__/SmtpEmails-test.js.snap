// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmtpEmails #sendResetPasswordEmail should sent email 1`] = `
Array [
  Object {
    "from": "me@gmail.com",
    "html": "<p>Hi John Smith,</p>

<p>Click the link below to set your password:</p>
<p><a href=\\"http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w\\">http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w</a></p>

<p>Thanks,<br>
CrossCloud team</p>
",
    "subject": "Set CrossCloud password",
    "text": "Hi John Smith,

Set your password here: http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w

Thanks,
CrossCloud team
",
    "to": "sample@gmail.com",
  },
]
`;

exports[`SmtpEmails #sendResetPasswordEmail should sent email 2`] = `
Array [
  Object {
    "from": "me@gmail.com",
    "html": "<p>Hi John Smith,</p>

<p>Click the link below to set your password:</p>
<p><a href=\\"http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w\\">http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w</a></p>

<p>Thanks,<br>
CrossCloud team</p>
",
    "subject": "Set CrossCloud password",
    "text": "Hi John Smith,

Set your password here: http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w

Thanks,
CrossCloud team
",
    "to": "sample@gmail.com",
  },
  Object {
    "from": "me@gmail.com",
    "html": "<p>Hi John Smith,</p>

<p>Click the link below to reset your password:</p>
<p><a href=\\"http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w\\">http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w</a></p>

<p>Thanks,<br>
CrossCloud team</p>
",
    "subject": "Reset CrossCloud password",
    "text": "Hi John Smith,

Reset your password here: http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w

Thanks,
CrossCloud team
",
    "to": "sample@gmail.com",
  },
]
`;
